### PR TITLE
Add GHC 9.10 support

### LINKED ISSUE #420: Support GHC 9.10
https://github.com/BNFC/bnfc/actions/runs/9127670093/job/25098437386?pr=482#step:20:25
> [__0] trying: doctest-0.22.2 (user goal)
> [__1] rejecting: ghc-9.10.1/installed-69c3 (conflict: doctest => ghc>=8.0 && <9.10)

### PR DESCRIPTION
**Pull request by sellout**
Fixes #420.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,6 +64,7 @@ jobs:
           # - 9.6.1
           - 9.6.2
           - 9.8.1
+          - 9.10.1
         include:
           - os: macos-latest
```
Comment (sol) [inline .github/workflows/build.yml:69]: This also needs to be changed to `macos-12`, as later versions of the runner image don't include `ghc` / `ghcup` anymore.

**Review by sol — CHANGES_REQUESTED** [review]

**Review by sol — CHANGES_REQUESTED** [review]
@sellout thanks for looking into this.  I think one test case is failing, if you can fix this (possibly needs, CPP, not sure), then this is good to merge.

**Comment by sellout** [comment]
Sorry for the delay here. I was running into other test failures and didn’t have time to dig in. But I just went over it again, and I was running the tests incorrectly 🤦

**Comment by sellout** [comment]
Sorry, fixing this with some broader local testing before I push again.

**Comment by sellout** [comment]
Ok, pushed a new version of the fix. I had had the wrong version in a conditional, so it worked with GHC 9.10, but I broke older versions. Tested 9.4, 9.6, 9.8, and 9.10 this time around.

**Review by sol — APPROVED** [review]

**Comment by sol** [comment]
Thanks 🎉

**Pull request merged**